src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage_StringIds.cs

```diff
@@ -72,19 +72,29 @@ internal partial class SQLitePersistentStorage
             // values.
             try
             {
-                stringId = connection.RunInTransaction(
+                // Pass in `throwOnSqlException: false` so we get the exception bubbled back to us as a result value.
+                var (result, exception) = connection.RunInTransaction(
                     static t => t.self.InsertStringIntoDatabase_MustRunInTransaction(t.connection, t.value),
-                    (self: this, connection, value));
+                    (self: this, connection, value),
+                    throwOnSqlException: false);
 
-                Contract.ThrowIfTrue(stringId == null);
-                return stringId;
-            }
-            catch (SqlException ex) when (ex.Result == Result.CONSTRAINT)
```

Comment:
this code used to try the operation, then specifically catch SqlException in the case of constraints.  That's simpler now by passing `throwOnSqlException: false`.  Now we just get the result-or-exception back and we can directly  check if it's a constraint violation below.